analytics/src/analytics/integrations/etldb/etldb.py

```diff
@@ -128,3 +130,18 @@ class EtlChangeType(Enum):
     NONE = 0
     INSERT = 1
     UPDATE = 2
+
+
+def etl_db_connection(func) -> Callable:
+    """Decorator to inject etlb connection into a wrapped function."""
+
+    @wraps(func)
+    def wrapper(*args: Any, **kwargs: Any):
+        etldb_conn = kwargs.get("etldb_conn")
+        if not etldb_conn:
+            etldb_conn = EtlDb()
+            kwargs.update({"etldb_conn": etldb_conn})
+
+        return func(*args, **kwargs)
+
+    return wrapper
```

Comment:
While this util is nice to have, I might suggest not doing it in this PR as it'll probably be 30+ lines of type issues to sort through.

I'd suggest just initializing the DB connection like other jobs do for now

Reply:
Removed. Main func now passes the connection to a helper func where the bulk of the code runs.